### PR TITLE
Improves documentation style for several classes

### PR DESCRIPTION
**Pull request by adlerhsieh**
- Uses third-person narrative.
- Uses asterisks for parameters.
- Adds backticks to class names.

### CODE REVIEW

File: src/set.cr [hunk 1]
```diff
@@ -155,7 +155,7 @@ struct Set(T)
     set
   end
 
-  # Difference: returns a new set containing elements in this set that are not
+  # Difference: returns a new `Set` containing elements in this set that are not
```
Comment (jhass) [inline src/set.cr]: I'd keep this at "set"

Comment (asterite) [inline src/set.cr]: Yes, me too. There's no point in having a link to the same type.

**Comment by adlerhsieh** [comment]
Thanks for advice. Updated.

**Comment by jhass** [comment]
Thanks!

**Pull request merged**